src/preprocessing/passes/non_clausal_simp_sat.cpp

```diff
@@ -0,0 +1,483 @@
+/*********************                                                        */
+/*! \file non_clausal_simp.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Aina Niemetz, Haoze Wu
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2018 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Non-clausal simplification preprocessing pass.
+ **
+ ** Run the nonclausal solver and try to solve all assigned theory literals.
+ **/
+
+ #include "preprocessing/passes/non_clausal_simp_sat.h"
+ #include <cryptominisat5/cryptominisat.h>
```

Comment:
The implementation needs to be protected by an `#ifdef CVC4_USE_CRYPTOMINISAT` otherwise compilation will fail when CVC4 is not configured with `--with-cryptominisat` because the headers are not available. In the case where we do not compile with CVC4, we should emit an error if the user tries to use the option `--skeleton-preprocessing`.